### PR TITLE
chore(CPlusPlus): add Reorder list

### PR DESCRIPTION
**Pull request by UG-SEP**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
Added Reorder List program in Algorithm/Linked-Lists/Reorder-List

**Other information:**

closes: #363

### CODE REVIEW

File: algorithms/CPlusPlus/Linked-Lists/Reorder-List.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    new_head=new_head->next;
+    slow->next=t->next;
+    t->next=slow;
+    prev=t->next;
+    t=t->next->next;
+    }
+    if(new_head!=NULL)
+    {
+      prev->next=new_head;
+    }
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Linked-Lists/Reorder-List.cpp:110]: fix the indentation here.

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

File: algorithms/CPlusPlus/Linked-Lists/Reorder-List.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    slow->next=t->next;
+    t->next=slow;
+    prev=t->next;
+    t=t->next->next;
+    }
+    if(new_head!=NULL)
+      {
+      prev->next=new_head;
+      }
+    }
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Linked-Lists/Reorder-List.cpp:113]: It still does not look right. Indent the code within the brackets

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

**Review by aayushjain7 — DISMISSED** [review]
Great Work! I guess the indent problem is some GitHub issue.

**Comment by Arsenic-ATG** [comment]
>  I guess the indent problem is some GitHub issue.

My guess is that it is due to inconsistent use of tabs and spaces in the program which are being differently rendered by OP's editor and the web-browser and ultimately messing up the indentation after pushing the changes to GitHub and being viewed from a different text editor.

This should be fixed after the commit [75a375](https://github.com/MakeContributions/DSA/pull/373/commits/75a375443c829a55ccbdd2615ffe8de1fed117b3)

**Comment by UG-SEP** [comment]
> > I guess the indent problem is some GitHub issue.
> 
> My guess is that it is due to inconsistent use of tabs and spaces in the program which are being differently rendered by OP's editor and the web browser, messing up the indentation after pushing the changes to GitHub and being viewed from a different text editor.
> 
> This should be fixed after the commit [75a375](https://github.com/MakeContributions/DSA/pull/373/commits/75a375443c829a55ccbdd2615ffe8de1fed117b3)

I have a solution for this @Arsenic-ATG and @aayushjain7  as me and @Arsenic-ATG know Qt so we can create an application that removes extra indent from the program. Hope you understand what I want to say

**Comment by Arsenic-ATG** [comment]
> > > I guess the indent problem is some GitHub issue.
> > 
> > 
> > My guess is that it is due to inconsistent use of tabs and spaces in the program which are being differently rendered by OP's editor and the web browser, messing up the indentation after pushing the changes to GitHub and being viewed from a different text editor.
> > This should be fixed after the commit [75a375](https://github.com/MakeContributions/DSA/pull/373/commits/75a375443c829a55ccbdd2615ffe8de1fed117b3)
> 
> I have a solution for this @Arsenic-ATG and @aayushjain7 as me and @Arsenic-ATG know Qt so we can create an application that removes extra indent from the program. Hope you understand what I want to say

I think the easiest method to fix it is to use a different text editor, and avoid mixing spaced and tabs with each other. Or if you are looking for an automated code formatter then you can use [this](https://codebeautify.org/cpp-formatter-beautifier) one.

**Review by aayushjain7 — APPROVED** [review]
Looks great now!! 💯

**Pull request merged**